spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/SparkWriteConf.java

```diff
@@ -304,4 +305,13 @@ public boolean caseSensitive() {
         .defaultValue(SQLConf.CASE_SENSITIVE().defaultValueString())
         .parse();
   }
+
+  public String branch() {
+    return confParser
+        .stringConf()
+        .option(SparkWriteOptions.BRANCH)
+        .sessionConf(SparkWriteOptions.BRANCH)
```

Comment:
Similar to above, I think this should be a `SparkSQLProperties` entry and should be something like `spark.sql.iceberg.write-branch`. Unlike the read side, it's correct to return this from `branch()` if it is defined because branches are automatically created.

Reply:
i am going to remove all session related changes from this PR so that can raise a follow up one. We can support only write by spark option in this PR.

Reply:
adding it back